spec/jobs/email_csv_job_spec.rb

```diff
@@ -0,0 +1,159 @@
+require "rails_helper"
+
+describe EmailCsvJob do
+  include Helpers
+
+  test_url = :test_url
+
+  let(:job) { described_class.new }
+  let(:user) { FactoryBot.create(:user) }
+  let(:organisation) { user.organisation }
+  let(:other_organisation) { FactoryBot.create(:organisation) }
+
+  context "when a log exists" do
+    let!(:lettings_log) do
+      FactoryBot.create(
+        :lettings_log,
+        owning_organisation: organisation,
+        managing_organisation: organisation,
+        ecstat1: 1,
+      )
+    end
+
+    let(:storage_service) { instance_double(Storage::S3Service) }
+    let(:mailer) { instance_double(CsvDownloadMailer) }
+
+    before do
+      FactoryBot.create(:lettings_log,
+                        :completed,
+                        owning_organisation: organisation,
+                        managing_organisation: organisation,
+                        created_by: user)
+
+      allow(Storage::S3Service).to receive(:new).and_return(storage_service)
+      allow(storage_service).to receive(:write_file)
+      allow(storage_service).to receive(:get_presigned_url).and_return(test_url)
+
+      allow(CsvDownloadMailer).to receive(:new).and_return(mailer)
+      allow(mailer).to receive(:send_email)
+    end
+
+    it "uses an appropriate filename in S3" do
+      expect(storage_service).to receive(:write_file).with(/logs-.*\.csv/, anything)
+      job.perform(user)
+    end
+
+    it "includes the organisation name in the filename when one is provided" do
+      expect(storage_service).to receive(:write_file).with(/logs-#{organisation.name}-.*\.csv/, anything)
+      job.perform(user, nil, {}, nil, organisation)
+    end
+
+    it "sends an E-mail with the presigned URL and duration" do
+      expect(mailer).to receive(:send_email).with(user, test_url, instance_of(Integer))
+      job.perform(user)
+    end
+
+    context "when writing to S3" do
+      before do
+        FactoryBot.create_list(:lettings_log, 4, owning_organisation: other_organisation)
+      end
+
+      def expect_csv
+        expect(storage_service).to receive(:write_file) do |_filename, data|
+          # Ignore BOM
```

Comment:
BOM?

Reply:
Byte Order Marker (from a fix Nat added to make Excel import the CSV as UTF-8 automatically). Will de-abbreviate in comment.